Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -1,5 +1,71 @@
+
+Next to the resources for which permission is requested, there is a set of capabilities, which can be required by the app developer in order to run the device. These capabilities (`UIRequiredDeviceCapabilities`) are listed at <TODO ADD REFERENCE TO https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/iPhoneOSKeys.html#//apple_ref/doc/uid/TP40009252-SW1 here!>. These capabilities are used by App Store and by iTunes to ensure that only compatible devices are listed. They often do not provide implicit permission (TODO: EXPLAIN WHAT IMPLICIT IS: NO USER INTERACTION THERE!). <TODO: CHECK FOR WHICH THIS IS DIFFERENT, SUCH AS INTER-AP COMMUNICATION>. Note that the actual available capabilities differ per type of developer profile used to sign the application (https://developer.apple.com/support/app-capabilities/).
+
+#### Static analysis
+
+Since iOS 10, there are three areas which you need to inspect for permssions:
+- the info.plist file,
+- the `<appname>.enttitlements` file, where <appname> is the name of the application, or the
+- the source-code.
+
+##### Info.plist
+The info.plist contains the texts offered to users when requesting permissioin to access the protected data or resources. The [Apple Documentation](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/requesting-permission/ "Requesting Permission") gives a clear instruction on how the user should be asked for permission to access the given resource. Following these guidelines should make it relatively simple to evaluate each and every entry in the info.plist file to check if the permission makes sense.
+For example, when you have a info.plist file, for a Solitair game which has, at least, the following content:
+
+```xml
+<key>NSHealthClinicalHealthRecordsShareUsageDescription</key>
+<string>Share your health data with us!</string>
+<key>NSCameraUsageDescription</key>
+<string>We want to access your camera</string>
+
+```
+Should be suspicious as a normal solitair game probably does not have any need for accessing the camera nor a user his health-records.
+Note that from iOS 10 onward you need to provide explanation in terms of these \*Description fields. See table 1-2 at the [Apple app programming guide](https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/ExpectedAppBehaviors/ExpectedAppBehaviors.html#//apple_ref/doc/uid/TP40007072-CH3-SW7 "Apple app programming guide") for a more complete overview of different keys to look for.
+
+<TODO: QUESTION: SHOULD I CREATE A COPY OF THE TABLE REPRESENTED AT THE LINKS ABOVE? OR JUST REFER TO THEM?>
```

Comment:
Just refer to them, otherwise we need to maintain it and have a duplicate. Let's always link instead of recreating the content, then we don't get outdated so fast

Reply:
👍